src/components/spreadsheet/equipment-table.js

```diff
@@ -0,0 +1,146 @@
+                params.rowIndex === 0 &&
+                params.node.rowPinned === 'top'
+            ) {
+                return {
+                    borderTop: '1px solid ' + theme.palette.primary.main,
+                    borderBottom: '1px solid ' + theme.palette.primary.main,
+                };
+            }
+        },
+        [
+            scrollToIndex,
+            theme.palette.primary.main,
+            theme.selectedRow.background,
+        ]
+    );
+
+    const getLocaleText = useCallback(
+        (params) => {
+            if (intl.locale !== LANG_FRENCH) {
+                return params.defaultValue;
+            } else {
+                return localeFrench.hasOwnProperty(params.key)
+                    ? localeFrench[params.key]
+                    : params.defaultValue;
+            }
+        },
+        [intl.locale]
+    );
+
+    const getRowId = useMemo(() => {
+        return (params) => params.data.id;
+    }, []);
+
+    //we filter enter key event to prevent closing or opening edit mode
+    const suppressKeyEvent = (params) => {
+        return !ALLOWED_KEYS.includes(params.event.key);
+    };
+
+    const defaultColDef = useMemo(
+        () => ({
+            filter: true,
+            sortable: true,
+            resizable: true,
+            lockPinned: true,
+            wrapHeaderText: true,
+            autoHeaderHeight: true,
+            suppressKeyboardEvent: (params) => suppressKeyEvent(params),
+        }),
+        []
+    );
+
+    return (
+        <>
+            {!fetched ? (
+                <LoaderWithOverlay
+                    color="inherit"
+                    loaderSize={70}
+                    loadingMessageText={'LoadingRemoteData'}
+                />
+            ) : (
+                <div className={clsx([theme.aggrid, classes.grid])}>
+                    <AgGridReact
+                        ref={gridRef}
+                        getRowId={getRowId}
+                        rowData={rows}
+                        pinnedTopRowData={editingData}
+                        getRowStyle={getRowStyle}
+                        columnDefs={columns}
+                        defaultColDef={defaultColDef}
+                        enableCellTextSelection={true}
+                        alwaysMultiSort={true}
+                        undoRedoCellEditing={true}
+                        editType={'fullRow'}
+                        onCellValueChanged={handleCellEditing}
+                        onRowValueChanged={handleRowEditing}
+                        onRowEditingStopped={handleEditingStopped}
+                        onColumnMoved={handleColumnDrag}
+                        suppressDragLeaveHidesColumns={true}
+                        suppressPropertyNamesCheck={true}
+                        suppressColumnVirtualisation={true}
+                        suppressClickEdit={true}
+                        getLocaleText={getLocaleText}
+                        context={{
```

Comment:
is it ok to always pass a new object here ?

Reply:
I guess it's better to put the object in a useMemo